### PR TITLE
Migrate Groups store to sync by default (GetGroups and Counts)

### PR DESCRIPTION
**Pull request by jespino**
#### Summary

Migrate Groups store to sync by default, functions:

- GetGroupsByChannel
- CountGroupsByChannel
- GetGroupsByTeam
- CountGroupsByTeam
- GetGroups

### CODE REVIEW

File: store/storetest/group_supplier.go [hunk 1]
```diff
@@ -1739,12 +1739,12 @@ func testGetGroupsByChannel(t *testing.T, ss store.Store) {
...
-			res := <-ss.Group().GetGroupsByChannel(tc.ChannelId, tc.Opts)
-			require.Nil(t, res.Err)
-			require.ElementsMatch(t, tc.Result, res.Data.([]*model.Group))
+			groups, err := ss.Group().GetGroupsByChannel(tc.ChannelId, tc.Opts)
+			require.Nil(t, err)
+			require.ElementsMatch(t, tc.Result, groups)
 			if tc.TotalCount != nil {
-				res = <-ss.Group().CountGroupsByChannel(tc.ChannelId, tc.Opts)
-				count := res.Data.(int64)
+				var count int64
```
Comment (mgdelacroix) [inline store/storetest/group_supplier.go]: nit: the signature of `CountGroupsByChannel` is already specifying that the return type is `int64`, is it neccessary to create the variable before the function call?

File: store/storetest/group_supplier.go [hunk 2]
```diff
@@ -1941,13 +1941,12 @@ func testGetGroupsByTeam(t *testing.T, ss store.Store) {
...
-			res := <-ss.Group().GetGroupsByTeam(tc.TeamId, tc.Opts)
-			require.Nil(t, res.Err)
-			groups := res.Data.([]*model.Group)
+			groups, err := ss.Group().GetGroupsByTeam(tc.TeamId, tc.Opts)
+			require.Nil(t, err)
 			require.ElementsMatch(t, tc.Result, groups)
 			if tc.TotalCount != nil {
-				res = <-ss.Group().CountGroupsByTeam(tc.TeamId, tc.Opts)
-				count := res.Data.(int64)
+				var count int64
```
Comment (mgdelacroix) [inline store/storetest/group_supplier.go]: Same here

File: store/storetest/group_supplier.go (hunk 1 above)
Comment (jespino) [reply store/storetest/group_supplier.go]: I did that to avoid the shadowing of the 'err' variable there. I'm just reusing the `err` variable.

Comment (jespino) [reply store/storetest/group_supplier.go]: With the `:=` notation can be a bit ambiguous, so, I prefer this way.

**Pull request merged**